### PR TITLE
Update 010_helpers.sql - Create procedure create_warehouses_with_sizes

### PR DESCRIPTION
**Pull request by scgkiran**
# Pull Request Template

## Description

Please include a summary of the change including relevant motivation and context.

## Checklist:

- [x] I have verified that my change is functionally correct (unit tests are great).
- [x] Queries against newly-added, user-facing objects can be expected to complete in 1's of seconds.
- [x] My change will not have a substantial increase the user's cost to have the app installed (did you add any new tasks?).

### CODE REVIEW

File: bootstrap/010_helpers.sql [hunk 1]
```diff
@@ -149,3 +149,80 @@ $$
...
+
+create or replace procedure tools.create_warehouses_with_sizes(prefix string, min_size string, max_size string)
+    returns string
+    language sql
+as
+begin
+    let min int := 0;
+    let max int := 0;
+    call warehouse_size_as_int(:min_size) INTO min;
+    call warehouse_size_as_int(:max_size) INTO max;
```
Comment (pravindra) [inline bootstrap/010_helpers.sql:183]: exception if max-min < 1 ?

File: bootstrap/010_helpers.sql [hunk 2]
```diff
@@ -149,3 +149,80 @@ $$
...
+            from values (\'X-Small\', 1), (\'Small\', 2), (\'Medium\', 3), (\'Large\', 4), (\'X-Large\', 5),
+            (\'2X-Large\', 6), (\'3X-Large\', 7), (\'4X-Large\', 8), (\'5X-Large\', 9), (\'6X-Large\', 10)
+            WHERE credit >= ? AND credit <= ?';
+    let rs resultset := (execute immediate sql using (min, max));
+
+    let wh_names string := '';
+    let c1 cursor for rs;
+    for record in c1 DO
+        let wh_name string := :prefix || '_' || record.size;
+        let cmd string := 'CREATE WAREHOUSE IF NOT EXISTS "' || wh_name || '" WITH WAREHOUSE_SIZE =\'' || record.size || '\' AUTO_SUSPEND = 60 INITIALLY_SUSPENDED = TRUE';
```
Comment (pravindra) [inline bootstrap/010_helpers.sql:195]: why 60 (and not 1) ?

File: bootstrap/010_helpers.sql (hunk 1 above)
Comment (pravindra) [reply bootstrap/010_helpers.sql:183]: also, if prefix is empty

File: bootstrap/010_helpers.sql [hunk 3]
```diff
@@ -149,3 +149,80 @@ $$
...
+    returns string
+    language sql
+as
+begin
+    let min int := 0;
+    let max int := 0;
+    call warehouse_size_as_int(:min_size) INTO min;
+    call warehouse_size_as_int(:max_size) INTO max;
+
+    let sql string := 'select column1 as size, column2 as credit
```
Comment (pravindra) [inline bootstrap/010_helpers.sql:185]: can we call this size_value or size_numeric instead ?

the credit values are different https://docs.snowflake.com/en/user-guide/warehouses-overview#warehouse-size

File: bootstrap/010_helpers.sql (hunk 1 above)
Comment (scgkiran) [reply bootstrap/010_helpers.sql:186]: fixed it

File: bootstrap/010_helpers.sql [hunk 4]
```diff
@@ -176,23 +176,32 @@ create or replace procedure tools.create_warehouses_with_sizes(prefix string, mi
     returns string
     language sql
 as
+declare
+    INVALID_WAREHOUSE_SIZE_PARAMS EXCEPTION(-20512, 'Warehouse min_size should be less than max_size');
+    INVALID_WAREHOUSE_PREFIX EXCEPTION(-20513, 'Warehouse predix cannot be empty');
```
Comment (pravindra) [inline bootstrap/010_helpers.sql:181]: nit: typo predix -> prefix

File: bootstrap/010_helpers.sql [hunk 5]
```diff
@@ -204,16 +213,25 @@ create or replace procedure tools.drop_warehouses_with_sizes(prefix string, min_
...
```
Comment (pravindra) [inline bootstrap/010_helpers.sql:218]: nit: predix -> prefix

**Review by pravindra — DISMISSED** [review]
lgtm but pls wait till ryan/josh have a look too.

**Review by rymurr — APPROVED** [review]

**Pull request merged**